docs/infra/upgrade-database.md

```diff
@@ -0,0 +1,70 @@
+# Upgrade database
+
+Upgrading the database between major versions (e.g., from Postgres 15 to 16) is a two-step process.
+
+1. Create a new DBParameterGroup for the new engine version and upgrade the database.
+2. Remove the old DBParamaterGroup for the prior engine version.
+
+These steps are a minimal starting point for the changes you'll need to make. As with any major change to your codebase, you should carefully test the impact of upgrading the database before applying it to a production environment. See also the AWS documentation for [Upgrading the PostgreSQL DB engine for Amazon RDS](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_UpgradeDBInstance.PostgreSQL.html#USER_UpgradeDBInstance.PostgreSQL.MajorVersion.Process).
+
+## 1. Creating a new DBParameterGroup and upgrading the database
+
+1. Set `allow_major_version_upgrade = true`
+
+Set this the `aws_rds_cluster` resource in [infra/modules/database/main.tf#L20](../../infra/modules/database/main.tf).
+
+2. (if needed) Update the `serverlessv2_scaling_configuration`
+
+Set the `min_capacity` to 4.0 (and adjust the `max_capacity` accordingly).
+If your minimum is lower than this, the upgrade will fail with `FATAL: shared memory segment sizes are configured too large`.
```

Comment:
Once committed, this documentation will be the [second reference ever to this error on Google](https://www.google.com/search?q=%22FATAL:+shared+memory+segment+sizes+are+configured+too+large%22&sca_esv=4d5fce8fe77ae59a&sca_upv=1&ei=3ExnZqanGfCs5NoPn4af-AY&ved=0ahUKEwimhbnk29GGAxVwFlkFHR_DB28Q4dUDCBE&uact=5&oq=%22FATAL:+shared+memory+segment+sizes+are+configured+too+large%22&gs_lp=Egxnd3Mtd2l6LXNlcnAiPSJGQVRBTDogc2hhcmVkIG1lbW9yeSBzZWdtZW50IHNpemVzIGFyZSBjb25maWd1cmVkIHRvbyBsYXJnZSJIAFAAWABwAHgBkAEAmAEAoAEAqgEAuAEDyAEA-AEBmAIAoAIAmAMAkgcAoAcA&sclient=gws-wiz-serp) 🙃